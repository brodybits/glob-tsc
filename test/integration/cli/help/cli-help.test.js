const execa = require('execa');

const path = require('path');

// THANKS for initial guidance:
// * https://medium.com/@ole.ersoy/unit-testing-commander-scripts-with-jest-bc32465709d6
// * https://github.com/superflycss/cli/blob/master/index.spec.js
// * https://github.com/brodybits/create-react-native-module/blob/0.11.0/tests/integration/cli/help/cli-help.test.js
test('bin/glob-tsc.js --help', async () => {
    const { stdout } = await execa.command(`node ${path.resolve('bin/glob-tsc.js')} --help`);

    expect(stdout).toMatchSnapshot();
});
