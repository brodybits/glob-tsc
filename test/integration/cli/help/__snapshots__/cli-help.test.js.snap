// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`bin/glob-tsc.js --help 1`] = `
"Usage:  glob-tsc [options]

Options:
  -V, --version               output the version number
  -f, --tsconfig-file <path>  tsconfig.json file location. Default 
                              ./tsconfig.json (default: \\"tsconfig.json\\")
  -g, --files-glob <globs>    File globs (default: [])
  -h, --help                  output usage information"
`;
